service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandler.java

```diff
@@ -103,4 +109,24 @@ private String getRequiredValue(Document xml, String xpath) {
         }
         return value;
     }
+
+    public EhrExtractStatus getEhrExtractStatus(GetGpcStructuredTaskDefinition structuredTaskDefinition) {
+        Optional<EhrExtractStatus> ehrStatus =
+            ehrExtractStatusRepository.findByConversationId(structuredTaskDefinition.getConversationId());
+        if (ehrStatus.isPresent()) {
+            return ehrStatus.get();
+        } else {
+            throw new RuntimeException("No EHR Extract Status for ConversationId: " + structuredTaskDefinition.getConversationId());
+        }
+    }
+
+    public void updateEhrExtractStatusAccessStructured(GetGpcStructuredTaskDefinition structuredTaskDefinition,
+        EhrExtractStatus ehrExtractStatus) {
+        var accessStructured = new EhrExtractStatus.GpcAccessStructured(ehrExtractStatus.getConversationId()
+            + "_gpc_structured.json", now(),
+            structuredTaskDefinition.getTaskId());
+        ehrExtractStatus.setGpcAccessStructured(accessStructured);
+        ehrExtractStatus.setUpdatedAt(now());
+        ehrExtractStatusRepository.save(ehrExtractStatus);
```

Comment:
This needs to be an update statement not a get/replace. What happens if two instance are updating this document at the same time? We don't have optimistic locking. The other party's updates would be lost!

Reply:
Discussed on call with @mattd-kainos  @mmackiewicz1 pending test